pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -478,6 +478,19 @@ in
 lib.extendDerivation
   validity.handled
   ({
+     # a recreation in Nix of the derivation produced by the cli when calling `nix develop`
```

Comment:
So this is duplicate code, which will go out of sync because of the distinct release cadence.
Furthermore, `nix develop` is already legacy code in my view, because of the architectural flaws outlined in https://github.com/NixOS/nix/issues/7468#issuecomment-1356401975 and related threads.

Reply:
> `nix develop` is not suitable for further development, pardon the pun. We should fix it first, by moving its implementation to stdenv, where it belongs. That change will subsume this PR in a way that's sustainable.

That is kinda the whole point of this PR. :sweat_smile: 

But seriously, even if we get a your new implementation setup really quickly we are still going to have users using versions with the old behavior for quite some time. I realize that it is duplicate code but since we can't use IFD there is not much we can do about it.